### PR TITLE
chore(mesh-layer): cleanup material parser textures

### PR DESCRIPTION
**Pull request by dryabinin-actionengine**
Explicitly remove textures in finalizeState to improve statistics calculation in luma.gl.
Also PR in luma.gl - https://github.com/visgl/luma.gl/pull/1633

### CODE REVIEW

File: modules/geo-layers/src/mesh-layer/mesh-layer.js [hunk 1]
```diff
@@ -124,6 +130,18 @@ export default class _MeshLayer extends SimpleMeshLayer {
 
     attribute.value = value;
   }
+
+  destroyMaterialParserTextures() {
+    if (this.state.materialParser) {
+      this.state.materialParser.destroyTextures();
+      this.setState({materialParser: null});
+    }
```
Comment (ibgreen) [inline modules/geo-layers/src/mesh-layer/mesh-layer.js:138]: Why not just call `destroy()` / `delete()` and create a new materialParser? I am not sure we even need this helper method.

```suggestion
    this.state.materialParser?.destroy();
    this.setState({materialParser: null});
```

Comment (dryabinin-actionengine) [reply modules/geo-layers/src/mesh-layer/mesh-layer.js]: done

**Review by ibgreen — APPROVED** [review]

File: modules/geo-layers/src/mesh-layer/mesh-layer.js [hunk 2]
```diff
@@ -98,15 +102,17 @@ export default class _MeshLayer extends SimpleMeshLayer {
     const unlit = Boolean(
       pbrMaterial.pbrMetallicRoughness && pbrMaterial.pbrMetallicRoughness.baseColorTexture
     );
-    const materialParser = new GLTFMaterialParser(this.context.gl, {
+
+    this.destroyMaterialParserTextures();
```
Comment (Pessimistress) [inline modules/geo-layers/src/mesh-layer/mesh-layer.js:106]: This helper method changes `parseMaterial()` from a pure function to something that mutates the layer state. Either set the new state here or just simply call `this.state.materialParser?.delete()`. Generally speaking it's a good idea not to rely on the caller to maintain `state.materialParser` when this method relies on it.

Comment (dryabinin-actionengine) [reply modules/geo-layers/src/mesh-layer/mesh-layer.js]: Ok, I'v kept only `this.state.materialParser?.delete()` to avoid mutation.

**Comment by dryabinin-actionengine** [comment]
@Pessimistress Please take a look at it again. It will be really good to have this change in `8.8.0-alpha.3` release with https://github.com/visgl/deck.gl/pull/6914 as well.

**Comment by dryabinin-actionengine** [comment]
@Pessimistress Updated with last master. I still waiting this PR in one of alpha versions. Please merge it and publish if everything is ok =)

File: modules/geo-layers/src/mesh-layer/mesh-layer.js (hunk 2 above)
Comment (Pessimistress) [reply modules/geo-layers/src/mesh-layer/mesh-layer.js]: Can you just remove the `destroyMaterialParserTextures` method and incorporate it in to `finalizeState`? It seems unnecessary to have a helper function for two lines of code that is only called once...

Comment (dryabinin-actionengine) [reply modules/geo-layers/src/mesh-layer/mesh-layer.js]: Done

**Review by Pessimistress — APPROVED** [review]

**Comment by Pessimistress** [comment]
Can you rebase master?

**Comment by dryabinin-actionengine** [comment]
> Can you rebase master?

Done.

**Comment by dryabinin-actionengine** [comment]
> TypeScript check is failing (`materialParser` needs to be added to the state type declaration)

Done. Also Added delete method to the luma `gltf-material-parser.` - https://github.com/visgl/luma.gl/pull/1650

**Pull request merged**